### DISCUSSION TITLE
Best Practices for Constraining ETH Amounts in Functions (Equivalent to ERC20 bound Usage)

### DISCUSSION THREAD

**Discussion by rasyidred** [Cyfrin/security-and-auditing-full-course-s23#88] [Q&A] [answered]
We have this snippet code from the lesson 5: T-Swap:

```javascript
function depositYeildERC20(uint256 _amount) public {
    uint256 amount = bound(_amount, 0, yeildERC20.balanceOf(owner));
    vm.startPrank(owner);
    yeildERC20.approve(address(handlerStatefulFuzzCatches), amount);
    handlerStatefulFuzzCatches.depositToken(yeildERC20, amount);
    vm.stopPrank();
}
```

In the context of the `depositYeildERC20` function, we use the bound function to limit the `_amount` input for ERC20 tokens. However, when working with native tokens like ETH, there is no explicit input parameter analogous to `_amount`.

Is there a recommended or best practice method to limit the amount sent with `msg.value` in a function, similar to how we constrain ERC20 amounts in the provided code snippet?

**Comment by johnsonstephan** [comment, marked as answer]
Hi @rasyidred, good question. 

When working with ETH, one technique to limit the amount sent with `msg.value` in a function is by using `require()`.

Here's an example, limiting the amount of ETH sent:

``` solidity
function depositETH() public payable {
   require(msg.value >= 0.1 ether && msg.value <= 1 ether, "Amount must be 0.1 - 1 ETH");
   // continue function 
}
```

**Reply by cromewar** [reply]
Nice answer @johnsonstephan

**Reply by rasyidred** [reply]
so there is no inbuilt function to limit the amount of ETH sent in Foundry yet. Thanks for the insight! @johnsonstephan
